### PR TITLE
Code-cleanup, bugfixes

### PR DESCRIPTION
**Pull request by pakar**
## PR Checklist

Please check all that apply to this PR using "x":

- [x] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [x] I have checked that this PR does not introduce a breaking change

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [x] Refactoring (no functional changes)
- [x] Other - Please describe: Cleanup of code

## Other information
While reviewing and trying to get a grip of the code-base i have made a few minor fixes and done some cleanup.

Do look a bit closer at the changes before merging.

### CODE REVIEW

**Review by kekkyojin — COMMENTED** [review]
FDroid build server seems to not be able to use Gradle 6.5. Currently, I am building LBRY Fdroid with 6.1.1 since the beginning.

The FDroid release could still use 6.1.1. Just commenting that it could be good to be able to use the same one, if it is going to be changed here.

**Comment by pakar** [comment]
@kekkyojin Good catch.. Changed to 6.1.1.

**Comment by kekkyojin** [comment]
Thanks, @pakar. I am right now testing to build it with 6.5 using the virtual machine which replicates the build server from FDroid, and it is working. When the virtual machine is created, the script is even adding 6.7. But, as I said, 6.1.1 works while also enabling new features on Android Studio.

If @akinwale is ok, the buildToolsVersion could also be updated. Currently, 29.0.1 is used, while FDroid cannot use it, so I have to 'sed' to change it into 29.0.2. It could be changed into a newer one, like 30.x.x. If it works on 'lbry-android' but there are problems on 'lbry-fdroid' I could again replace it with a version which works on their build server.

**Comment by kekkyojin** [comment]
Thanks, @pakar1. When this is merged, I will remove the prebuild step on the FDroid build script.

**Comment by pakar** [comment]
@kekkyojin @akinwale I'm done for now with this round of cleanup.

File: app/build.gradle [hunk 1]
```diff
@@ -14,11 +14,10 @@ android {
 
     defaultConfig {
         applicationId "io.lbry.browser"
-        minSdkVersion 21
-        targetSdkVersion 29
+        minSdkVersion 24
+        targetSdkVersion 30
```
Comment (akinwale) [inline app/build.gradle:18]: These need to be left as they are. `targetSdkVersion` should only be changed if absolutely necessary (required by Google Play). `minSdkVersion` indicates the minimum supported Android version (which is 5.0, and we don't intend to stop supporting that yet).

File: app/build.gradle [hunk 2]
```diff
@@ -3,8 +3,8 @@ import com.google.gms.googleservices.GoogleServicesPlugin
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion 29
-    buildToolsVersion "29.0.1"
+    compileSdkVersion 30
+    buildToolsVersion "29.0.2"
```
Comment (akinwale) [inline app/build.gradle:7]: These values can be left as they are. We only have to change them if absolutely necessary as they would require modifications to the Docker image used for the automated CI builds.

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -12,6 +12,7 @@
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
     <uses-permission android:name="com.android.vending.BILLING" />
     <uses-permission android:name="android.permission.WAKE_LOCK"/>
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
```
Comment (akinwale) [inline app/src/main/AndroidManifest.xml:15]: The app does not require this permission.

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,6 +1,6 @@
-#Fri Apr 03 15:21:09 WAT 2020
+#Tue Jan 12 20:03:24 CET 2021
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.1.1-all.zip
```
Comment (akinwale) [inline gradle/wrapper/gradle-wrapper.properties:6]: Please revert the Gradle version back to 5.6.4. Again, we don't need to change anything that affects the automated builds unless absolutely necessary.

**Review by akinwale — CHANGES_REQUESTED** [review]
Hey, thank you for the code cleanup. Everything looks good except for the issues I raised regarding the automated build system.

**Comment by akinwale** [comment]
@pakar @pakar1 Will you be able to make the requested changes so that we can merge your changes?

**Comment by pakar** [comment]
@akinwale I have not yet had the time to do any changes for those things.

The reason for updating the sdk versions is due to the current code is using features not available in previous sdk versions. Before i would switch back the the original (target/compile etc) versions i would need to add checks in the code to prevent failing on older phones. (see commit comments for those updates)

ACCESS_NETWORK_STATE is used by ExoPlayer.

For the gradle-bump this was mainly kept so the FDroid build uses the same version. See comment from @kekkyojin . This one i don't mind reverting.

**Comment by kekkyojin** [comment]
ACCESS_NETWORK_STATE is used but not required. I agree it should be removed from this PR.

**Comment by akinwale** [comment]
https://github.com/lbryio/lbry-android/pull/1164

**Pull request closed without merging**